packages/components/inputs/localized-rich-text-input/src/rich-text-input.tsx

```diff
@@ -35,6 +35,7 @@ type TRichTextInputProps = {
   toggleLanguage: (language: string) => void;
   warning?: ReactNode;
   error?: string;
+  additionalInfo?: ReactNode;
```

Comment:
There's a type mismatch here.
The `additionalInfo` prop the `LocalizedRichTextInput` component receives has this type:
```ts
Record<
    string,
    | string
    | ReactNode
    | (MessageDescriptor & {
        values: Record<string, ReactNode>;
      })
  >;
```

This means that, if the parent component receives a message descriptor, we won't be rendering the right message.

I don't see specs testing what happens when we pass message descriptions as part of the `additionalInfo` prop so perhaps we can add them and use them to make sure we implement that code path.

Reply:
I adjusted that and have a question. How do I fix that type mismatch which happens after doing your requested change? I have to admit I barely have done typescript

`Type 'ReactNode | (MessageDescriptor & { values: Record<string, ReactNode>; })' is not assignable to type 'Record<string, ReactNode | (MessageDescriptor & { values: Record<string, ReactNode>; })> | undefined'.
  Type 'null' is not assignable to type 'Record<string, ReactNode | (MessageDescriptor & { values: Record<string, ReactNode>; })> | undefined'.`

Reply:
Hi Lucas,

I've just pushed [this commit](https://github.com/commercetools/ui-kit/pull/2710/commits/85b31144a828f7769027cdcc7b5f7eb3194f05a4) to fix the issue.

The problem is that the outer components receive a `Record` (an object where keys are locales and values are the message in potentially different format) but the underlying components expect only the message.

I also refactored a bit the `AdditionalInfoMessage` so its easier for consumers to provide its value.

Please take a look at the changes in the commit and let me know if there are ok for you.

Also, remember it would be great if we could add a test in the input components to validate they work when a message descriptor or a React element are provided 🙏

Reply:
I will follow up with those tests 🙇‍♂️